core/src/builder.spec.ts

```diff
@@ -167,15 +167,13 @@ describe('building', () => {
       });
 
       describe('when replacing fields', () => {
-        describe('when field does not exist', () => {
-          it('should not add the desired fields', () => {
+        describe('when field exists', () => {
+          it('should replace the desired fields', () => {
             const transformers = {
               graphql: Transformer<TestUser, TestUser>('graphql', {
-                // The `identifier` is not part of the `TestUser`, so TS yells at us as expected.
-                // However for the purpose of the test, we ignore the type check.
-                // @ts-ignore
                 replaceFields: ({ fields }) => ({
-                  identifier: fields.id,
+                  ...fields,
+                  id: fields.id + 1,
```

Comment:
The `replaceFields` now should return the fully final transformed object, not just a partial representation.

This is important because the result of a transform does not necessarily need to be an object, but can also be an array, a string, a number, etc.

Reply:
Sorry, why can't I just mutate? Cause otherwise I can't mutate primitives?

Reply:
What do you mean by "mutate"?

Reply:
Instead of returning something new just mutating it. `field.id = field.id + 1` then `return field`?

Reply:
Ah you mean just in this test example?

Reply:
I think I just misunderstood the initial concern. It's fine that the whole field has to be returned not just a partial of it. If it's then being mutated or if it's an shallowly immutable clone doesn't matter.